docs/book/user-guide/advanced-guide/pipelining-features/managing-steps.md

```diff
@@ -300,3 +300,46 @@ If you do not want to store the logs in your artifact store, you can:
 2. Disable it by using the environmental variable `ZENML_DISABLE_STEP_LOGS_STORAGE` and setting it to `true`. This environmental variable takes precedence over the parameters mentioned above.
 
 <figure><img src="https://static.scarf.sh/a.png?x-pxid=f0b4f458-0a54-4fcd-aa95-d5ee424815bc" alt="ZenML Scarf"><figcaption></figcaption></figure>
+
+
+## Allow step retry in case of failure
+
+ZenML provides a built-in retry mechanism that allows you to configure automatic retries for your steps in case of failures. This can be useful when dealing with intermittent issues or transient errors. A common pattern when trying to run a step on GPU-backed hardware is that the provider will not have enough resources available, so you can set ZenML to handle the retries until the resources free up. You can configure three parameters for step retries:
+
+* **max_retries:** The maximum number of times the step should be retried in case of failure.
+* **delay:** The initial delay in **seconds** before the first retry attempt.
+* **backoff:** The factor by which the delay should be multiplied after each retry attempt.
+
+### Using the @step decorator:
+
+```python
+from zenml.config.retry_config import StepRetryConfig
+
+@step(
+    retry=StepRetryConfig(
+        max_retries=3, 
+        delay=10, 
+        backoff=2
+    )
+)
+def my_step() -> None:
+    raise Exception("This is a test exception")
+```
+
+You can also configure step retry using the [Configuration YAML](https://docs.zenml.io/user-guide/production-guide/configure-pipeline#breaking-down-our-configuration-yaml).
+Just like the Pythonic version we set the retry config in a `steps` section where we specify the name of the step and the retry configuration.
+### Using a configuration file (e.g., config.yaml):
+
+```yaml
```

Comment:
```suggestion
### Using the @step decorator:

You can specify the retry configuration directly in the definition of your step as follows:

```python
from zenml.config.retry_config import StepRetryConfig

@step(
    retry=StepRetryConfig(
        max_retries=3, 
        delay=10, 
        backoff=2
    )
)
def my_step() -> None:
    raise Exception("This is a test exception")
```

### Using a configuration file (e.g., config.yaml):

You can also configure step retry using the [Configuration YAML](https://docs.zenml.io/user-guide/production-guide/configure-pipeline#breaking-down-our-configuration-yaml).
Just like the Pythonic version we set the retry config in a `steps` section where we specify the name of the step and the retry configuration.

```yaml
```